src/v2/System/useFeatureFlag.tsx

```diff
@@ -35,5 +38,50 @@ export function useFeatureVariant(featureName: string): Variant | null {
     return null
   }
 
+  // detremine if we should track the experimentView
+  if (typeof window !== "undefined") {
+    let trackFeatureView = maybeTrackFeatureView(featureName, variant.name)
+
+    if (trackFeatureView) {
+      trackEvent({
+        action: Schema.ActionType.ExperimentViewed,
+        service: "unleash",
+        experiment_name: featureName,
+        variant_name: variant.name,
+        payload: variant.payload,
+      })
+    }
+  }
+
   return variant
 }
+
+function maybeTrackFeatureView(
+  featureName: string,
+  variantName: string
+): boolean {
+  const experimentName = `${featureName}+${variantName}`
+  let viewedExperiments = getExperimentsViewed()
+
+  if (viewedExperiments.includes(experimentName)) {
+    return false
+  }
+
+  viewedExperiments.push(experimentName)
+  setExperimentsViewed(viewedExperiments)
+
+  return true
+}
+
+function getExperimentsViewed(): string[] {
+  let experimentsViewed = window.localStorage.getItem("experimentsViewed")
+
+  return experimentsViewed === null ? [] : JSON.parse(experimentsViewed)
+}
+
+function setExperimentsViewed(_viewedExperiments: string[]) {
```

Comment:
`_someVariableName` (the underscore) signifies that a variable is unused

Reply:
probably the linter complained for shadowing and that's the reason for the underscore. I'd just ignore the rule 🤔

Reply:
better to just name it something else than to bypass strong built-in conventions